python/constellation/satellites/LeCroy/LeCroySatellite.py

```diff
@@ -92,12 +88,54 @@ def do_run(self, payload: Any) -> str:
                     event_payload = np.append(event_payload, trg_offsets)
                     event_payload = np.append(event_payload, wave_array)
                 self.data_queue.put((event_payload.tobytes(), {"dtype": f"{event_payload.dtype}"}))
+            except socket.timeout:
+                self.log.warning("Timeout encountered while retrieving the sequence.")
+                continue
             except (socket.error, struct.error) as e:
                 self.log.error(str(e))
                 self._scope.clear()
                 continue
-            num_events_acquired += self._num_sequences
+            self._num_triggers_acquired += self._num_sequences
             num_sequences_acquired += 1
-            self.log.info(f"Fetched event {num_events_acquired}/sequence {num_sequences_acquired}")
+            self.log.info(f"Fetched event {self._num_triggers_acquired}/sequence {num_sequences_acquired}")
 
+        self.EOR["current_time"] = datetime.datetime.now().timestamp()
         return "Finished acquisition"
+
+    def do_stopping(self) -> str:
+        self._scope.clear()
+        self.log.info(f"Stopping the run after {self._num_triggers_acquired} event(s)")
+        return "Stopped acquisition"
+
+    @cscp_requestable
+    def get_num_triggers(self, request: CSCP1Message) -> [str, int, dict[str, Any]]:
+        if self.fsm.current_state_value == SatelliteState.RUN:
+            return f"Number of triggers: {self._num_triggers_acquired}", self._num_triggers_acquired, {}
+        return "Not running", -1, {}
+
+    @schedule_metric("", MetricsType.LAST_VALUE, 10)
+    def NUM_TRIGGERS(self) -> int | None:
+        if self.fsm.current_state_value == SatelliteState.RUN:
+            return self._num_triggers_acquired
+        return None
```

Comment:
@simonspa correct, but this returns `None` if not in `RUN` so everything is fine

Reply:
Indeed, this is where the implementations of the metric and the custom command are diverging. And it makes me realise there is a typo in the README about the list of supported states for these two!